Цель работы: реализовать базовый анализ датасета: вывести весь датасет и описать его, вывести размерность массива, вывести названия всех колонок, вывести уникальные значения, отсортировать по определенным параметрам, удалить ненужные столбцы, заменить определенные значения в датасете, удалить дубликаты, алализ с помощью функций info и describe, провести выборку данных по строкам и столбцам с помощью loc, сохраниить новый датасет.

In [3]:
import pandas as pd

выводим датасет

In [6]:
genshin = pd.read_csv(r"C:\Users\User\Downloads\archive (1)\Genshin_Impact_All_Character_Stat.csv")

genshin

Character  Lv  Rarity Element    Weapon Main role  Ascension  Base HP  \
0       Amber   1       4    Pyro       Bow   Sub DPS        ATK      793   
1       Amber  20       4    Pyro       Bow   Sub DPS        ATK     2038   
2       Amber  20       4    Pyro       Bow   Sub DPS        ATK     2630   
3       Amber  40       4    Pyro       Bow   Sub DPS        ATK     3940   
4       Amber  40       4    Pyro       Bow   Sub DPS        ATK     4361   
..        ...  ..     ...     ...       ...       ...        ...      ...   
569    Kokomi  70       5   Hydro  Catalyst    Healer  Hydro DMG    10306   
570    Kokomi  70       5   Hydro  Catalyst    Healer  Hydro DMG    10945   
571    Kokomi  80       5   Hydro  Catalyst    Healer  Hydro DMG    11885   
572    Kokomi  80       5   Hydro  Catalyst    Healer  Hydro DMG    12524   
573    Kokomi  90       5   Hydro  Catalyst    Healer  Hydro DMG    13471   

     Base ATK  Base DEF  
0          19        50  
1          48       129  
2          62       167  
3          93       250  
4         103       277  
..        ...       ...  
569       179       503  
570       190       534  
571       207       580  
572       218       611  
573       234       657  

[574 rows x 10 columns]

вывелся весь датасет

выводим размерность датасета

In [7]:
genshin.shape

(574, 10)

вывелся размер датасета, 574 - длина, 10 - ширина

выводим наименования колонок

In [8]:
genshin.columns

Index(['Character', 'Lv', 'Rarity', 'Element', 'Weapon', 'Main role',
       'Ascension', 'Base HP', 'Base ATK', 'Base DEF'],
      dtype='object')

получилось десять названий всех колонок

выводим уникальные значения колонок

In [10]:
for column in genshin.columns:
    unique_values = genshin[column].unique()
    print(f'{column}: {unique_values[:30]}')


Character: ['Amber' 'Barbara' 'Beidou' 'Bennett' 'Chongyun' 'Diluc' 'Fischl' 'Jean'
 'Kaeya' 'Keqing' 'Klee' 'Lisa' 'Mona' 'Ningguang' 'Noelle' 'Qiqi' 'Razor'
 'Sucrose' 'Venti' 'Xiangling' 'Xiao' 'Xingqiu' 'Tartaglia' 'Zhongli'
 'Diona' 'Xinyan' 'Ganyu' 'Albedo' 'Rosaria' 'Ayaka']
Lv: [ 1 20 40 50 60 70 80 90]
Rarity: [4 5]
Element: ['Pyro' 'Hydro' 'Electro' 'Cryo' 'Anemo' 'Geo' 'All']
Weapon: ['Bow' 'Catalyst' 'Claymore' 'Sword' 'Polearm']
Main role: ['Sub DPS' 'Healer' 'DPS' 'Support']
Ascension: ['ATK' 'HP' 'Electro DMG' 'Energy Recharge' 'CRIT Rate' 'Healing Bonus'
 'CRIT DMG' 'Pyro DMG' 'Elemental Mastery' 'Geo DMG' 'DEF' 'Physical DMG'
 'Anemo DMG' 'Hydro DMG' 'Cryo DMG']
Base HP: [ 793 2038 2630 3940 4361 5016 5578 6233 6654 7309 7730 8385 8806 9461
  821 2108 2721 4076 4512 5189 5770 6448 6884 7561 7996 8674 9110 9787
 1094 2811]
Base ATK: [ 19  48  62  93 103 118 131 147 157 172 182 198 208 223  13  34  44  66
  73  84  94 105 112 123 130 141 148 159  63 104]
Base DEF: [ 50 1

вывелись все уникальные значения из всех колонок

сортируем персонажей по редкости и оружию 

In [11]:
sorted_genshin = genshin.sort_values(by=["Rarity", "Weapon"])

выводим отсортированный датасет

In [12]:
print(sorted_genshin)

    Character  Lv  Rarity Element Weapon Main role Ascension  Base HP  \
0       Amber   1       4    Pyro    Bow   Sub DPS       ATK      793   
1       Amber  20       4    Pyro    Bow   Sub DPS       ATK     2038   
2       Amber  20       4    Pyro    Bow   Sub DPS       ATK     2630   
3       Amber  40       4    Pyro    Bow   Sub DPS       ATK     3940   
4       Amber  40       4    Pyro    Bow   Sub DPS       ATK     4361   
..        ...  ..     ...     ...    ...       ...       ...      ...   
513  Traveler  70       5     All  Sword   Support       ATK     8401   
514  Traveler  70       5     All  Sword   Support       ATK     8885   
515  Traveler  80       5     All  Sword   Support       ATK     9638   
516  Traveler  80       5     All  Sword   Support       ATK    10122   
517  Traveler  90       5     All  Sword   Support       ATK    10875   

     Base ATK  Base DEF  
0          19        50  
1          48       129  
2          62       167  
3          93      

вывелся датасет, рассортированный по редкости и оружию

удаляем ненужный столбец Main role

In [13]:
genshin = genshin.drop(["Main role"], axis=1)

выводим датасет после удаления столбца

In [14]:
print(genshin)

    Character  Lv  Rarity Element    Weapon  Ascension  Base HP  Base ATK  \
0       Amber   1       4    Pyro       Bow        ATK      793        19   
1       Amber  20       4    Pyro       Bow        ATK     2038        48   
2       Amber  20       4    Pyro       Bow        ATK     2630        62   
3       Amber  40       4    Pyro       Bow        ATK     3940        93   
4       Amber  40       4    Pyro       Bow        ATK     4361       103   
..        ...  ..     ...     ...       ...        ...      ...       ...   
569    Kokomi  70       5   Hydro  Catalyst  Hydro DMG    10306       179   
570    Kokomi  70       5   Hydro  Catalyst  Hydro DMG    10945       190   
571    Kokomi  80       5   Hydro  Catalyst  Hydro DMG    11885       207   
572    Kokomi  80       5   Hydro  Catalyst  Hydro DMG    12524       218   
573    Kokomi  90       5   Hydro  Catalyst  Hydro DMG    13471       234   

     Base DEF  
0          50  
1         129  
2         167  
3         2

вывелся датасет без колонки Main role

заменяем значения элемента piro на fire

In [15]:
genshin['Element'] = genshin['Element'].replace('Pyro', 'Fire')

выводим измененный датасет

In [16]:
print(genshin)

    Character  Lv  Rarity Element    Weapon  Ascension  Base HP  Base ATK  \
0       Amber   1       4    Fire       Bow        ATK      793        19   
1       Amber  20       4    Fire       Bow        ATK     2038        48   
2       Amber  20       4    Fire       Bow        ATK     2630        62   
3       Amber  40       4    Fire       Bow        ATK     3940        93   
4       Amber  40       4    Fire       Bow        ATK     4361       103   
..        ...  ..     ...     ...       ...        ...      ...       ...   
569    Kokomi  70       5   Hydro  Catalyst  Hydro DMG    10306       179   
570    Kokomi  70       5   Hydro  Catalyst  Hydro DMG    10945       190   
571    Kokomi  80       5   Hydro  Catalyst  Hydro DMG    11885       207   
572    Kokomi  80       5   Hydro  Catalyst  Hydro DMG    12524       218   
573    Kokomi  90       5   Hydro  Catalyst  Hydro DMG    13471       234   

     Base DEF  
0          50  
1         129  
2         167  
3         2

вывелся датасет, где элемент piro быз заменен на fire

удаляем дубликаты

In [17]:
genshin.drop_duplicates(inplace=True)

выводим датасет

In [18]:
print(genshin)

    Character  Lv  Rarity Element    Weapon  Ascension  Base HP  Base ATK  \
0       Amber   1       4    Fire       Bow        ATK      793        19   
1       Amber  20       4    Fire       Bow        ATK     2038        48   
2       Amber  20       4    Fire       Bow        ATK     2630        62   
3       Amber  40       4    Fire       Bow        ATK     3940        93   
4       Amber  40       4    Fire       Bow        ATK     4361       103   
..        ...  ..     ...     ...       ...        ...      ...       ...   
569    Kokomi  70       5   Hydro  Catalyst  Hydro DMG    10306       179   
570    Kokomi  70       5   Hydro  Catalyst  Hydro DMG    10945       190   
571    Kokomi  80       5   Hydro  Catalyst  Hydro DMG    11885       207   
572    Kokomi  80       5   Hydro  Catalyst  Hydro DMG    12524       218   
573    Kokomi  90       5   Hydro  Catalyst  Hydro DMG    13471       234   

     Base DEF  
0          50  
1         129  
2         167  
3         2

вывелся датасет без дубликатов

анализ с помощью info

In [19]:
genshin.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 574 entries, 0 to 573
Data columns (total 9 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   Character  574 non-null    object
 1   Lv         574 non-null    int64 
 2   Rarity     574 non-null    int64 
 3   Element    574 non-null    object
 4   Weapon     574 non-null    object
 5   Ascension  574 non-null    object
 6   Base HP    574 non-null    int64 
 7   Base ATK   574 non-null    int64 
 8   Base DEF   574 non-null    int64 
dtypes: int64(5), object(4)
memory usage: 40.5+ KB


info выводит общую информацию о датасете, включая количество непустых значений, типы данных каждого столбца и потребление памяти.

анализ с помощью describe

In [20]:
genshin.describe()

Lv      Rarity       Base HP    Base ATK    Base DEF
count  574.000000  574.000000    574.000000  574.000000  574.000000
mean    52.214286    4.512195   6860.010453  148.527875  423.303136
std     24.996567    0.500287   3325.201592   77.011853  202.560931
min      1.000000    4.000000    763.000000    8.000000   48.000000
25%     40.000000    4.000000   4405.000000   91.250000  276.000000
50%     55.000000    5.000000   6942.500000  148.000000  431.000000
75%     70.000000    5.000000   9280.000000  200.750000  575.000000
max     90.000000    5.000000  15552.000000  349.000000  922.000000

describe вычисляет основные статистические показатели для каждого числового столбца в датасете, такие как среднее значение, стандартное отклонение, минимальное и максимальное значения, медиану и квартили.

с помощью  loc выберем строки, где значение в столбце 'Element' равно 'Fire', и только столбцы 'Character' и 'Lv

In [21]:
selected_genshin = genshin.loc[genshin['Element'] == 'Fire', ['Character', 'Lv']]

выводим датасет

In [23]:
print(selected_genshin)

    Character  Lv
0       Amber   1
1       Amber  20
2       Amber  20
3       Amber  40
4       Amber  40
..        ...  ..
485   Yoimiya  70
486   Yoimiya  70
487   Yoimiya  80
488   Yoimiya  80
489   Yoimiya  90

[126 rows x 2 columns]


вывелся датасет, который выводит только пиро персонажей

сохраняем датасет

In [24]:
selected_genshin.to_csv("selected_genshin.csv", index=False)

Вывод:в ходе практической работы удалось реализовать базовый анализ датасета: вывести весь датасет и описать его, вывести размерноть массива, наименование всех колонок, уникальные значения отсортировать по определенным параметрам, удалить ненужные столбцы, заменить определенные значения в датасете, удалить дубликаты, провести анализ с помощью функций info и describe, провести выборку данных по строкам и столбцам с помощью loc, сохранить новый датасет.